src/storage/UpdateEdgeProcessor.cpp

```diff
@@ -0,0 +1,382 @@
+        key_ = NebulaKeyUtils::edgeKey(partId, edgeKey.src, edgeKey.edge_type,
+                                       edgeKey.ranking, edgeKey.dst, now);
+        auto schema = this->schemaMan_->getEdgeSchema(spaceId_, edgeKey.edge_type);
+        RowWriter writer(schema);
+        edgeReader_ = RowReader::getRowReader(writer.encode(), schema);
+        filterReader_ = RowReader::getRowReader(writer.encode(), schema);
+    } else {
+        VLOG(3) << "Missed partId " << partId << ", edge " << edgeKey.src << " --> " << edgeKey.dst;
+        return kvstore::ResultCode::ERR_KEY_NOT_FOUND;
+    }
+    this->collectProps(edgeReader_.get(), key_, props, fcontext, collector);
+    return ret;
+}
+
+
+kvstore::ResultCode UpdateEdgeProcessor::processEdge(PartitionID partId,
+                                                     const cpp2::EdgeKey& edgeKey) {
+    insertEdge_ = false;
+    cpp2::UpdateRespData updateRespData;
+    updateRespData.set_upsert(false);
+    updateRespData.set_edge_key(edgeKey);
+    FilterContext fcontext;
+    RowWriter writer(nullptr);
+    PropsCollector collector(&writer);
+    auto ret = collectEdgesProps(partId,
+                                 edgeKey,
+                                 this->edgeContext_.props_,
+                                 &fcontext,
+                                 &collector);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        return ret;
+    }
+    for (auto& tc : this->tagContexts_) {
+        VLOG(3) << "partId " << partId << ", vId " << edgeKey.src
+                << ", tagId " << tc.tagId_ << ", prop size " << tc.props_.size();
+        ret = this->collectVertexProps(partId,
+                                       edgeKey.src,
+                                       tc.tagId_,
+                                       tc.props_,
+                                       &fcontext,
+                                       &collector);
+        if (ret != kvstore::ResultCode::SUCCEEDED) {
+            return ret;
+        }
+    }
+
+    // the Where clause for filting
+    auto& getters = expCtx_->getters();
+    getters.getSrcTagProp = [&, this] (const std::string& tagName,
+                                       const std::string& prop) -> VariantType {
+        auto it = fcontext.tagFilters_.find(std::make_pair(tagName, prop));
+        if (it == fcontext.tagFilters_.end()) {
+            return std::string("Unknown");
+        }
+        VLOG(1) << "Hit srcProp for tag " << tagName << ", prop "
+                << prop << ", value " << it->second;
+        return it->second;
+    };
+    getters.getEdgeProp = [&, this] (const std::string &prop) -> VariantType {
+        auto res = RowReader::getPropByName(this->filterReader_.get(), prop);
+        CHECK(ok(res));
+        auto val = value(std::move(res));
+        VLOG(1) << "Hit edgeProp for edge prop " << prop << ", value " << val;
+        return val;
+    };
+    if (this->exp_ != nullptr) {
```

Comment:
maybe filter_ is more readable.

Reply:
filterReader_  rename to  filter_?

Reply:
sorry, I mean exp_ rename to filter_

Reply:
The `exp_` is the member of QueryBaseProcessor rather than UpdateEdgeProcessor. I try not to modify other code.